### DISCUSSION TITLE
Vitest failed in GitHub Action, but works in local

### DISCUSSION THREAD

**Discussion by fityannugroho** [vitest-dev/vitest#6007] [Q&A]
I have test my code using vitest and mock the http request with [msw](https://mswjs.io). All tests passed in local, but when I push it, it's failed in CI that using GitHub Action. I have tried to change the vitest's `pool` into **forks** but it doesn't works. You can see the detail in this PR: https://github.com/fityannugroho/img-dl/pull/14. FYI, I'm using **Vitest 1.6.0**.

This is the last error message from the CI test:
```
> img-dl@0.5.1 test
> vitest run


 RUN  v1.6.0 /home/runner/work/img-dl/img-dl

 ✓ test/downloader.spec.ts  (64 tests) 141ms
 ❯ test/index.spec.ts  (7 tests | 1 failed) 635ms
   ❯ test/index.spec.ts > `imgdl` > should download multiple images if array of URLs is provided
     → promise rejected "Error: ENOENT: no such file or directory,… { …(4) }" instead of resolving
 ✓ test/cli.spec.ts  (10 tests) 3762ms

⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  test/index.spec.ts > `imgdl` > should download multiple images if array of URLs is provided
AssertionError: promise rejected "Error: ENOENT: no such file or directory,… { …(4) }" instead of resolving
 ❯ test/index.spec.ts:112:39
    110|         path: path.resolve(directory, `img-${i + 1}.jpg`),
    111|       });
    112|       await expect(fs.access(img.path)).resolves.not.toThrow();
       |                                       ^
    113|     }
    114|   });

Caused by: Caused by: Error: ENOENT: no such file or directory, access '/home/runner/work/img-dl/img-dl/test/tmp/img-1.jpg'
 ❯ test/index.spec.ts:112:7

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯


Error: AssertionError: promise rejected "Error: ENOENT: no such file or directory,… { …(4) }" instead of resolving
 ❯ test/index.spec.ts:112:39

Caused by: Error: ENOENT: no such file or directory, access '/home/runner/work/img-dl/img-dl/test/tmp/img-1.jpg'
 ❯ test/index.spec.ts:112:7


 Test Files  1 failed | 2 passed (3)
      Tests  1 failed | 80 passed (81)
   Start at  12:37:21
   Duration  4.16s (transform 170ms, setup 0ms, collect 900ms, tests 4.54s, environment 1ms, prepare 295ms)

Error: Process completed with exit code 1.
```

I don't know what caused it, either the `vitest`, `msw`, or the GitHub Action.
Can someone assist me?

**Comment by AriPerkkio** [comment]
> Caused by: Caused by: Error: ENOENT: no such file or directory, access '/home/runner/work/img-dl/img-dl/test/tmp/img-1.jpg'

Where does this file come from? Are you sure CI is able to write that file? Does the test work locally when you pass `CI=true`, e.g. `CI=true npm run test`?

**Reply by fityannugroho** [reply]
The file should be the result of a successful image download.

And I have tried to run the test with `CI=true` in local, and it also passed.

![Screenshot 2024-07-01 165056](https://github.com/vitest-dev/vitest/assets/63894003/29f3f9d9-8ac9-45b8-987f-35a631a194f0)

Is there any GitHub Action policy that might make CI unable to write new files?
